plugins/uefi/fu-uefi-bootmgr.c

```diff
@@ -273,14 +274,17 @@ fu_uefi_bootmgr_bootnext (const gchar *esp_path,
 	secure_boot = fu_efivar_secure_boot_enabled ();
 	if (secure_boot) {
 		/* test to make sure shim is there if we need it */
-		shim_app = fu_uefi_get_esp_app_path (esp_path, "shim", error);
+		if (fu_device_get_metadata_boolean (device, "RequireRemovablePath"))
+			shim_app = fu_uefi_get_esp_app_path (device, esp_path, "boot", error);
+		else
+			shim_app = fu_uefi_get_esp_app_path (device, esp_path, "shim", error);
```

Comment:
Here's an untested alternative proposal that would meet my suggestion.

```
g_autoptr(GError) error_local = NULL;
shim_app = fu_uefi_get_esp_app_path (device, esp_path, "shim", &error_local);

if (shim_app == NULL) {
        if (fu_device_get_metadata_boolean (device, "RequireRemovablePath")) {
                shim_app = fu_uefi_get_esp_app_path (device, esp_path, "boot", error);
        else
                g_propagate_error (error, g_steal_pointer (&error_local));
        if (shim_app == NULL)
                return FALSE;
}
```

Reply:
And in doing that, you probably should rename `RequireRemovablePath` to `FallbacktoRemovablePath`